### PR TITLE
add menu transfer uv paste invert normals

### PR DESCRIPTION
**Pull request by maxRobinot**
Allows user to transfer UVs from one mesh to another when this one has been mirrored on one axis in edit mode and normals reverted to "outside".

### CODE REVIEW

**Comment by mifth** [comment]
Hi.
Could you give us blend file with example of the mirror?

File: uv_magic_uv/muv_transuv_ops.py [hunk 1]
```diff
@@ -87,58 +87,75 @@ class MUV_TransUVPaste(bpy.types.Operator):
...
 
-                # parse all faces according to selection history
-                all_sorted_faces = main_parse(self, active_obj, bm, uv_layer, sel_faces, active_face)
 
-                if all_sorted_faces:
-                    # check amount of copied/pasted faces
-                    if len(all_sorted_faces) != len(props.topology_copied):
-                        self.report({'WARNING'}, "Mesh has different amount of faces!!")
-                        return {'CANCELLED'}
+def MUV_TransUVPasteExecute(self, context, invert_normals):
```
Comment (nutti) [inline uv_magic_uv/muv_transuv_ops.py]: Can you rename this function as ```execute_paste()``` ?
Functions are named by snake case (words are separated by ```_```) in this add-on.

**Comment by nutti** [comment]
Awesome contribution!!
I would like to try your feature if you have mirrored example too.

**Comment by mifth** [comment]
@nutti i can check the code. If you want.
There could be some optimizations.

**Comment by nutti** [comment]
@mifth 

Thanks.
Please optimize this if you can spare your time.
I'm developing features for v4.0, so I can't spare my time to check this code well.

**Comment by maxRobinot** [comment]
Sorry for the ";", didn't see them.

Here is an example where it doesn't work with the original paste but works with the added one:
[example_mirrored.blend.zip](https://github.com/nutti/Magic-UV/files/121305/example_mirrored.blend.zip)

**Comment by mifth** [comment]
HI.
Did you copy/paste the code twice? I mean Paste UV code of your new class.

**Comment by maxRobinot** [comment]
Hi,

I'm not sure to understand what you meant.
The code in function execute_paste() is a copy/paste of former function MUV_TransUVPaste().
I did this to not duplicate the code because the only difference is the new parameter of function main_parse().

**Comment by mifth** [comment]
I mean these lines https://github.com/dwarflabs/Magic-UV/commit/34337f740cd65c8daf1a4bde28563fd6b20d3281#diff-c31e70513a04ee2cb59053553d0598a1R132
132-155 lines. Are they copied from other class? So you have the same code in 2 classes?

**Comment by maxRobinot** [comment]
The class MUV_TransUVPaste and MUV_TransUVPasteInvertNormals are quasi similar.
The difference is in their execute() function where I pass 1 or -1 in parameter of execute_paste().

**Comment by mifth** [comment]
Ah, sorry i see. Yes, there is no double code. 

What about to create a checkbox in history like we did at PasteSelectionSequence? http://i.imgur.com/nSb8q91.png

It will be much easier and much less modification.

**Comment by maxRobinot** [comment]
I did with the checkbox, it is much less modification indeed.

**Comment by mifth** [comment]
Hi 

@maxRobinot  I did some changes to your commit and made at as simple as it possible. I hope you don't mind.
I changed some CANCELLED tweaks to FINISHED. Ir will allow us to see the checkbox during pasting.

Here is my new pull request: https://github.com/nutti/Magic-UV/pull/39
Here is video test: https://youtu.be/3j1ulKpMwVU

@nutti I guess you can commit my pull request. if everything is fine.

**Comment by maxRobinot** [comment]
I committed an hour ago with a version similar with the checkbox, maybe you didn't see it.
But it's fine, you did it the same way as me, so we can keep your version.

**Comment by mifth** [comment]
@maxRobinot i made it based on your commit. I tweaked some things only so that all worked well. )
And I put your name at the file.

**Comment by mifth** [comment]
@maxRobinot  and i changed only one file.

**Comment by maxRobinot** [comment]
Ok, thanks.

**Pull request merged**